### PR TITLE
chore(core): Add a CLI flag to allow for empty configs

### PR DESCRIPTION
**Pull request by bruceg**
This allows for users to set up an empty config that may later be replaced with actual running components. With this option enabled, Vector will not immediately shut down with an empty config, but it will follow the normal shutdown path when components are present.

### CODE REVIEW

**Review by jszwedko — COMMENTED** [review]
Do we need to add this to the documentation?

**Comment by bruceg** [comment]
> Do we need to add this to the documentation?

Yeah, I'll add that to the CLI reference right away. I had mistakenly assumed anything needed would end up being auto-generated from the doc comments.

File: website/cue/reference/cli.cue [hunk 1]
```diff
@@ -644,6 +648,12 @@ cli: {
 				"""
 			type: bool: default: false
 		}
+		VECTOR_ALLOW_EMPTY_CONFIG: {
+			description: """
+				Allow the configuration to run without any components. This is useful for loading in an empty stub config that will later be replaced with actual components. Note that this is likely not useful without also watching for config file changes as described for `--watch-empty`.
```
Comment (neuronull) [inline website/cue/reference/cli.cue:653]: **🥜 nitpick:** ‏


```suggestion
				Allow the configuration to run without any components. This is useful for loading in an empty stub config that will later be replaced with actual components. Note that this is likely not useful without also watching for config file changes as described in `--watch-empty`.
```

**Review by neuronull — APPROVED** [review]
LGTM just one nitpicky thing

File: website/cue/reference/cli.cue (hunk 1 above)
Comment (jszwedko) [reply website/cue/reference/cli.cue:653]: I don't think `--watch-empty` is a config option 🤔

Comment (bruceg) [reply website/cue/reference/cli.cue]: Fixed in 7ac746a22

Comment (bruceg) [reply website/cue/reference/cli.cue]: Ugh, fixed again in 77c1c2e02

**Review by jhgilbert — APPROVED** [review]

**Review by jszwedko — APPROVED** [review]
We had some discussion about whether we should add an option for this or just make it the default behavior but there were enough concerns about users being surprised by this change that we decided to make it opt-in for now. If we get more of a signal that this is better default behavior in the future we can deprecate and switch the default.

**Pull request merged**